brainframe_qt/util/oauth/reply_handler.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from PyQt5.QtCore import QObject, pyqtSignal
+from PyQt5.QtNetwork import QHostAddress
+from PyQt5.QtNetworkAuth import QOAuthHttpServerReplyHandler
+
+from brainframe_qt.util.oauth.base import AuthTokenResponse
+
+
+class PKCEReplyHandler(QOAuthHttpServerReplyHandler):
+    ready = pyqtSignal()
+    """Emitted when the ReplyHandler is ready for connections"""
+    completed = pyqtSignal(AuthTokenResponse)
+    """Emitted when the ReplyHandler has received an Authorization Code"""
+
+    _PORTS = [21849, 32047, 31415]
```

Comment:
A comment here might be good. Something like "These are the ports that the OAuth2 client is configured to redirect requests to. We supply multiple options in case one is already being used by another service."